### PR TITLE
feat(core): add edgelessTheme property and edgelessDefault theme setting

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1430 AF-1471

https://github.com/user-attachments/assets/d997ac6c-ce94-4fa4-ab34-29b36c7796ea

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#8614** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8614?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/hooks/use-journal.ts [hunk 1]
```diff
@@ -11,11 +11,30 @@ import {
...
+  currentAppTheme: string
+) => {
+  switch (defaultTheme) {
+    case 'dark':
+      return 'dark';
+    case 'light':
+      return 'light';
+    case 'specified':
+      return currentAppTheme === 'dark' ? 'dark' : 'light';
+    case 'auto':
```
Comment (forehalo) [inline packages/frontend/core/src/components/hooks/use-journal.ts:32]: 'auto' => 'system'

File: packages/frontend/component/src/components/theme-provider/index.tsx [hunk 2]
```diff
@@ -1,12 +1,27 @@
-import { ThemeProvider as NextThemeProvider } from 'next-themes';
+import { LiveData } from '@toeverything/infra';
+import { ThemeProvider as NextThemeProvider, useTheme } from 'next-themes';
 import type { PropsWithChildren } from 'react';
+import { useEffect } from 'react';
 
 const themes = ['dark', 'light'];
 
+export const appTheme$ = new LiveData<string | undefined>(undefined);
```
Comment (pengx17) [inline packages/frontend/component/src/components/theme-provider/index.tsx]: Put this value into framework

File: packages/frontend/component/src/components/theme-provider/index.tsx [hunk 3]
```diff
@@ -1,12 +1,27 @@
...
+
+export function ThemeObserver() {
+  const { resolvedTheme } = useTheme();
+
+  useEffect(() => {
+    appTheme$.value = resolvedTheme;
```
Comment (pengx17) [inline packages/frontend/component/src/components/theme-provider/index.tsx]: appTheme$.emit(resolvedTheme)

File: packages/frontend/component/src/components/theme-provider/index.tsx (hunk 2 above)
Comment (akumatus) [reply packages/frontend/component/src/components/theme-provider/index.tsx]: Thanks for your advice! I added a `AppThemeService` and regist it on the framework.

File: packages/frontend/component/src/components/theme-provider/index.tsx (hunk 3 above)
Comment (akumatus) [reply packages/frontend/component/src/components/theme-provider/index.tsx]: Fixed. Thanks!

**Review by pengx17 — APPROVED** [review]

**Review by forehalo — CHANGES_REQUESTED** [review]
use stable bs

**Review by forehalo — APPROVED** [review]

**Review by akumatus — APPROVED** [review]

**Pull request merged**